File: dataset/python2java/python/COUNT_BINARY_DIGIT_NUMBERS_SMALLER_N.py
def f_gold ( N ) :
    q = collections . deque ( )
    q.append ( 1 )
    cnt = 0
    while ( q ) :
        t = q.popleft ( )
        if ( t <= N ) :
            cnt = cnt + 1
            q.append ( t * 10 )
            q.append ( t * 10 + 1 )
    return cnt

File: result/translate/starchat/python2java/java/COUNT_BINARY_DIGIT_NUMBERS_SMALLER_N.java
public static int f_gold ( int N ) {
        Deque<Integer> q = new ArrayDeque<>();
        q.addLast(1);
        int cnt = 0;
        while (!q.isEmpty()) {
            int t = q.removeFirst();
            if (t <= N) {
                cnt++;
                q.addLast(t * 10);
                q.addLast(t * 10 + 1);
            }
        }
        return cnt;
    }